docs/relay-selector.md

```diff
@@ -49,40 +49,27 @@ Endpoints may be filtered by:
 Whilst all user selected constraints are always honored, when the user hasn't selected any specific
 constraints, following default ones will take effect:
 
-- If no tunnel protocol is specified, the first three connection attempts will use WireGuard. All
-  remaining attempts will use OpenVPN. If no specific constraints are set:
-  - The first two attempts will connect to a Wireguard server, first on a random port, and then port
-    53.
-  - The third attempt will connect to a Wireguard server on port 80 with _udp2tcp_.
-  - Remaining attempts will connect to OpenVPN servers, first over UDP on two random ports, and then
-    over TCP on port 443. Remaining attempts alternate between TCP and UDP on random ports.
+- The first four connection attempts will use Wireguard.
+  - The first two of these attempts will connect to a Wireguard relay on a random port.
```

Comment:
Are both of these identical attempts represented by this part of `RETRY_ORDER`?
```rust
// 0
RelayQueryBuilder::new().build(),
// 1
RelayQueryBuilder::new().wireguard().build(),
```
I guess since wireguard is default, `RelayQueryBuilder::new().build()` will result in a wireguard relay if possible, but that is not evident from just the definition of `RETRY_ORDER`. This documentation also doesn't ever say that it tries to connect over opvenVPN with a random port. Perhaps the first two items should be replaced by 
```rust
// 0
RelayQueryBuilder::new().wireguard().build(),
// 1
RelayQueryBuilder::new().openvpn().build(),
```
and the corresponding text with something like
"The first two attempts will connect to a Wireguard relay on a random port, followed by an openVPN relay on a random port."

What do you think?

Reply:
Not specifying any tunnel protocol implies that Wireguard will be used, so even if it looks a bit funky I would say that the document matches the implementation in this case. However, according to the spec that I'm implementing, the second retry attempt should be Wireguard on port 443, so I have implemented the wrong spec :D Easily fixed though, and thanks for making me aware of this!

I agree that we should try OpenVPN on a random port _eventually_, but it is also not part of the spec :shrug: Let's discuss it IRL!